src/electron/electron/main/services/WindowService.ts

```diff
@@ -307,39 +345,65 @@ export class WindowService {
     const trayImage = nativeImage.createFromPath(appIcon);
     trayImage.setTemplateImage(true);
     this.tray = new Tray(trayImage);
-    this.updateTray();
+    await this.updateTray();
   }
 
-  private getTrayMenuTemplate(): MenuItemConstructorOptions[] {
-    const versionAndUpdate: MenuItemConstructorOptions[] = [
+  private async getTrayMenuTemplate(): Promise<MenuItemConstructorOptions[]> {
+    const settings: Settings = await Settings.findOne({ where: { onlyOneEntityShouldExist: 1 } });
+    
+    const versionUpdateAndStudyInfo: MenuItemConstructorOptions[] = [
       { label: `Version ${app.getVersion()}`, enabled: false },
       {
         label: 'Check for updates',
         enabled: false,
         click: () => this.appUpdaterService.checkForUpdates({ silent: false })
       },
-      { type: 'separator' }
+      { type: 'separator' },
+      { 
+        label: 'Copy Subject Id',
+        click: () => clipboard.writeText(settings.subjectId)  
+      },
+      {
+        label: `Subject ID: ${settings.subjectId}`,
+        enabled: false, 
+      }
     ];
+
+    if (studyConfig.displayDaysParticipated) {
```

Comment:
i think we could simplify this, instead of the condition and then pushing the item to the list, we could add:

`visible: studyConfig.displayDaysParticipated`

(similarly below for the "open task planning"